dbsim/db_threads.cpp

```diff
@@ -256,6 +235,30 @@ namespace
         bool detached_;
     };
 
+    thread_local ti_thread* this_ti_thread = nullptr;
+
+    static const wsrep::thread_service::thread_key* main_thread_key = nullptr;
+    struct main_thread_initializer
+    {
+      main_thread_initializer() {
+          main_thread_key
+              = reinterpret_cast<const wsrep::thread_service::thread_key*>(
+                  append_key("main", "thread"));
+          this_ti_thread = new ti_thread(main_thread_key);
```

Comment:
I suppose such a memory allocation from the global variable may cause memory sanitizer or a checked allocator to falsely complain about a memory leak.

If that's the case, maybe something like that would work:
```
static bool main_thread_initializer()
{
      const wsrep::thread_service::thread_key* main_thread_key
              = reinterpret_cast<const wsrep::thread_service::thread_key*>(
                  append_key("main", "thread"));
      static ti_thread main_thread(main_thread_key);
      this_ti_thread = &main_thread;
      return true;
}
static bool main_thread_init = main_thread_initializer();
```

Also it looks like `main_thread_key` is not needed to be static.